### PR TITLE
very naive test for generic int

### PR DESCRIPTION
**Pull request by nemo83**

### CODE REVIEW

**Comment by benwaffle** [comment]
Does this code run atoi on the string with the quotes included?

**Comment by nemo83** [comment]
I'll reply inline

File: utils/utils.go [hunk 1]
```diff
@@ -5,13 +5,45 @@ import (
...
+// that has either a number or a string value.
+// E.g. if the json field contains an string "42", the
+// FlexInt value will be "42".
+type FlexInt int
+
+// UnmarshalJSON implements the json.Unmarshaler interface, which
+// allows us to ingest values of any json type as an int and run our custom conversion
+
+func (fi *FlexInt) UnmarshalJSON(b []byte) error {
+   if b[0] != '"' {
```
Comment (nemo83) [inline utils/utils.go]: if the first byte of the field is not a `"` than it should be an int and casts it directly. @benwaffle

File: utils/utils.go [hunk 2]
```diff
@@ -5,13 +5,45 @@ import (
...
+          return json.Unmarshal(b, (*int)(fi))
+   }
+   var s string
+   if err := json.Unmarshal(b, &s); err != nil {
+          return err
+   }
```
Comment (nemo83) [inline utils/utils.go]: or there is a `"` so it must be a string...

File: utils/utils.go [hunk 3]
```diff
@@ -5,13 +5,45 @@ import (
...
+   i, err := strconv.Atoi(s)
```
Comment (nemo83) [inline utils/utils.go]: now parse the string (at this point there should be no `"`)

**Pull request merged**